tests/unittests/tests-color/tests-color.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2016 Cenk Gündoğan <mail@cgundogan.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+#include <errno.h>
+#include <stdint.h>
+
+#include "embUnit/embUnit.h"
+
+#include "color.h"
+
+#include "tests-color.h"
+
+static void test_str2rgb__success(void)
+{
+    const char *color_str = "#F09A1D";
+    color_argb_t color;
+
+    TEST_ASSERT_EQUAL_INT(0, color_str2rgb(color_str, &color));
+    TEST_ASSERT_EQUAL_INT(0xF0, color.argb.r);
+    TEST_ASSERT_EQUAL_INT(0x9A, color.argb.g);
+    TEST_ASSERT_EQUAL_INT(0x1D, color.argb.b);
+}
+
+static void test_str2rgb__fail(void)
+{
+    const char *color_str = "F09A1D";
+    color_argb_t color;
+
+    TEST_ASSERT_EQUAL_INT(-1, color_str2rgb(color_str, &color));
```

Comment:
Why is it required? Can't you save a few byte of ROM by just ignoring every non hex-char?